Exercise 1
# **Introduction**
L’analyse des données est une méthode qui permet d’étudier des informations afin d’en tirer des conclusions utiles. Aujourd’hui, les entreprises, les banques, les hôpitaux et même les écoles produisent beaucoup de données. Ces données doivent être analysées pour aider à prendre de meilleures décisions.

# 1. Qu’est-ce que l’analyse de données ?
L’analyse de données consiste à collecter, organiser, nettoyer et interpréter des données. Son objectif est de comprendre une situation, découvrir des tendances ou résoudre un problème.

Par exemple, une entreprise peut analyser ses ventes pour savoir quels produits sont les plus achetés, à quel moment et par quels clients.
# 2. Pourquoi l’analyse des données est-elle importante ?
L’analyse des données est importante parce qu’elle permet de prendre des décisions basées sur des faits réels. Elle aide aussi à gagner du temps, réduire les erreurs et améliorer les performances.

Dans le monde moderne, les organisations utilisent les données pour mieux comprendre leurs clients, prévoir les risques, améliorer leurs services et rester compétitives.

#3. Domaines d’application
a) La santé

Dans la santé, l’analyse de données permet de suivre les patients, prévoir certaines maladies et améliorer la gestion des hôpitaux.

b) La finance

Dans les banques, elle aide à détecter les fraudes, analyser les risques et sécuriser les transactions.

c) Le commerce

Dans le commerce, elle permet de comprendre les habitudes des clients, améliorer les ventes et proposer de meilleurs produits.

#Conclusion

L’analyse des données est devenue essentielle dans le monde actuel. Elle transforme les données en informations utiles pour mieux décider, mieux prévoir et mieux agir. Elle est utilisée dans plusieurs domaines comme la santé, la finance et le commerce.

In [ ]:
import pandas as pd

Exercise 2


In [ ]:
sleep_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Time Americans Spend Sleeping.csv")
sleep_data.head()
sleep_data.info()
sleep_data.describe()
sleep_data.isnull().sum()

mental_health_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Mental health Depression disorder Data.csv")
mental_health_data.head()
mental_health_data.info()
mental_health_data.describe()
mental_health_data.isnull().sum()

credit_card_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\crx.csv")
credit_card_data.head()
credit_card_data.info()
credit_card_data.describe()
credit_card_data.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    object 
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    object 
 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB


/tmp/ipykernel_5223/2718386.py:8: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  mental_health_data = pd.read_csv("/content/Mental health Depression disorder Data.csv")


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108553 entries, 0 to 108552
Data columns (total 11 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   index                      108553 non-null  int64  
 1   Entity                     108553 non-null  object 
 2   Code                       103141 non-null  object 
 3   Year                       108553 non-null  object 
 4   Schizophrenia (%)          25875 non-null   object 
 5   Bipolar disorder (%)       19406 non-null   object 
 6   Eating disorders (%)       100236 non-null  object 
 7   Anxiety disorders (%)      6468 non-null    float64
 8   Drug use disorders (%)     6468 non-null    float64
 9   Depression (%)             6468 non-null    float64
 10  Alcohol use disorders (%)  6468 non-null    float64
dtypes: float64(4), int64(1), object(6)
memory usage: 9.1+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 689 entries, 0 to 688
Data columns (

,0
b,0
30.83,0
0,0
u,0
g,0
w,0
v,0
1.25,0
t,0
t.1,0


Exercise 3

1. Dataset : Sommeil des Américains

Ce dataset permet d’étudier les habitudes de sommeil des Américains. D’après l’exemple Kaggle, ce jeu de données contient 945 lignes et 9 colonnes

In [ ]:
sleep_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Time Americans Spend Sleeping.csv")
sleep_data.head()

,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


# Conclusion
Dans ce dataset, les colonnes numériques comme index,Year,vg hrs per day sleeping,Standard Error et Age Group sont quantitatives. Les colonnes qui décrivent Period, Type of Days,Activity et Sex sont qualitatives.

2. Dataset : Tendances des troubles de santé mentale

Ce dataset étudie les troubles de santé mentale de l'Afghanistan, comme la dépression, la schizophrénie ou d’autres troubles mentaux.

In [ ]:
mental_health_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Mental health Depression disorder Data.csv")
mental_health_data.head()

/tmp/ipykernel_5223/1483970859.py:1: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  mental_health_data = pd.read_csv("/content/Mental health Depression disorder Data.csv")


,index,Entity,Code,Year,Schizophrenia (%),Bipolar disorder (%),Eating disorders (%),Anxiety disorders (%),Drug use disorders (%),Depression (%),Alcohol use disorders (%)
0,0,Afghanistan,AFG,1990,0.16056,0.697779,0.101855,4.828830,1.677082,4.071831,0.672404
1,1,Afghanistan,AFG,1991,0.160312,0.697961,0.099313,4.829740,1.684746,4.079531,0.671768
2,2,Afghanistan,AFG,1992,0.160135,0.698107,0.096692,4.831108,1.694334,4.088358,0.670644
3,3,Afghanistan,AFG,1993,0.160037,0.698257,0.094336,4.830864,1.705320,4.096190,0.669738
4,4,Afghanistan,AFG,1994,0.160022,0.698469,0.092439,4.829423,1.716069,4.099582,0.669260


Conclusion

Dans ce dataset, les colonnes comme le pays ou le code sont qualitatives. Les colonnes contenant des taux, des pourcentages ou des valeurs de troubles mentaux sont quantitatives.

3. Dataset : Approbations de cartes de crédit

Ce dataset contient des informations sur des demandes de cartes de crédit. Il vient d’une version nettoyée du dataset UCI sur les approbations de cartes de crédit.

In [ ]:
credit_card_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\crx.csv")
credit_card_data.head()

,b,30.83,0,u,g,w,v,1.25,t,t.1,01,f,g.1,00202,0.1,+
0,a,58.67,4.460,u,g,q,h,3.04,t,t,6,f,g,00043,560,+
1,a,24.50,0.500,u,g,q,h,1.50,t,f,0,f,g,00280,824,+
2,b,27.83,1.540,u,g,w,v,3.75,t,t,5,t,g,00100,3,+
3,b,20.17,5.625,u,g,w,v,1.71,t,f,0,f,s,00120,0,+
4,b,32.08,4.000,u,g,m,v,2.50,t,f,0,t,g,00360,0,+


Conclusion

Dans ce dataset, les colonnes liées	30.83,	0,1.25,01,	00202 et	0.1 sont quantitatives. Les colonnes b,u,g,w,v,t,t.1,f et g.1 sont qualitatives.

Exercise 4

In [ ]:

iris_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Iris.csv")

iris_data.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


In [ ]:

types_iris = pd.DataFrame({
    "Nom_colonne": [
        "Id",
        "SepalLengthCm",
        "SepalWidthCm",
        "PetalLengthCm",
        "PetalWidthCm",
        "Species"
    ],
    "Catégorie": [
        "Qualitative",
        "Quantitative",
        "Quantitative",
        "Quantitative",
        "Quantitative",
        "Qualitative"
    ],
    "Justification": [
        "C’est un identifiant. Même s’il est numérique, il sert seulement à identifier chaque ligne.",
        "C’est la longueur du sépale en centimètres. C’est une mesure numérique.",
        "C’est la largeur du sépale en centimètres. C’est une valeur mesurable.",
        "C’est la longueur du pétale en centimètres. C’est une mesure numérique.",
        "C’est la largeur du pétale en centimètres. C’est une mesure numérique.",
        "C’est l’espèce de la fleur. C’est une catégorie."
    ]
})

types_iris

,Nom_colonne,Catégorie,Justification
0,Id,Qualitative,"C’est un identifiant. Même s’il est numérique,..."
1,SepalLengthCm,Quantitative,C’est la longueur du sépale en centimètres. C’...
2,SepalWidthCm,Quantitative,C’est la largeur du sépale en centimètres. C’e...
3,PetalLengthCm,Quantitative,C’est la longueur du pétale en centimètres. C’...
4,PetalWidthCm,Quantitative,C’est la largeur du pétale en centimètres. C’e...
5,Species,Qualitative,C’est l’espèce de la fleur. C’est une catégorie.


Exercise 5

In [ ]:

sleep_data = pd.read_csv("C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\Time Americans Spend Sleeping.csv")

sleep_data.head()
sleep_data.columns

Index(['index', 'Year', 'Period', 'Avg hrs per day sleeping', 'Standard Error',
       'Type of Days', 'Age Group', 'Activity', 'Sex'],
      dtype='object')

In [ ]:
analyse_colonnes = pd.DataFrame({
    "Colonne": [
        "Year",
        "Period",
        "Avg hrs per day sleeping",
        "Standard Error",
        "Type of Days",
        "Age Group",
        "Activity",
        "Sex"
    ],
    "Type_analyse_possible": [
        "Analyse de tendance",
        "Analyse de période",
        "Analyse principale du sommeil",
        "Analyse de fiabilité",
        "Comparaison de groupes",
        "Comparaison de groupes",
        "Analyse par activité",
        "Comparaison de groupes"
    ],
    "Justification": [
        "Permet d'observer l'évolution du temps moyen de sommeil selon les années.",
        "Permet de savoir à quelle période les données correspondent.",
        "C'est la variable principale du dataset, car elle indique le nombre moyen d'heures de sommeil par jour.",
        "Permet d'évaluer la précision ou la marge d'erreur liée à la moyenne du sommeil.",
        "Permet de comparer le sommeil entre les jours ouvrables, les week-ends ou autres types de jours.",
        "Permet de comparer le sommeil selon les tranches d'âge.",
        "Permet d'analyser le sommeil selon le type d'activité observée.",
        "Permet de comparer le sommeil entre les hommes et les femmes."
    ]
})

analyse_colonnes

,Colonne,Type_analyse_possible,Justification
0,Year,Analyse de tendance,Permet d'observer l'évolution du temps moyen d...
1,Period,Analyse de période,Permet de savoir à quelle période les données ...
2,Avg hrs per day sleeping,Analyse principale du sommeil,"C'est la variable principale du dataset, car e..."
3,Standard Error,Analyse de fiabilité,Permet d'évaluer la précision ou la marge d'er...
4,Type of Days,Comparaison de groupes,Permet de comparer le sommeil entre les jours ...
5,Age Group,Comparaison de groupes,Permet de comparer le sommeil selon les tranch...
6,Activity,Analyse par activité,Permet d'analyser le sommeil selon le type d'a...
7,Sex,Comparaison de groupes,Permet de comparer le sommeil entre les hommes...


Exercise 6

In [ ]:

donnees = pd.DataFrame({
    "Source de données": [
        "Les rapports financiers d'une entreprise sont stockés dans un fichier Excel",
        "Photographies téléchargées sur une plateforme de médias sociaux",
        "Une collection d'articles de presse sur un site web",
        "Données d'inventaire dans une base de données relationnelle",
        "Enregistrements d'entretiens issus d'une étude de marché"
    ],
    "Type": [
        "Structurées",
        "Non structurées",
        "Non structurées",
        "Structurées",
        "Non structurées"
    ],
    "Justification": [
        "Les données sont organisées en lignes et colonnes dans un fichier Excel.",
        "Les photographies sont des images, elles ne sont pas organisées sous forme de tableau.",
        "Les articles sont du texte libre, donc ils ne suivent pas toujours une structure fixe.",
        "Les données sont stockées dans des tables avec des colonnes bien définies.",
        "Les entretiens sont souvent des fichiers audio ou du texte libre, donc non structurés."
    ]
})

donnees

,Source de données,Type,Justification
0,Les rapports financiers d'une entreprise sont ...,Structurées,Les données sont organisées en lignes et colon...
1,Photographies téléchargées sur une plateforme ...,Non structurées,"Les photographies sont des images, elles ne so..."
2,Une collection d'articles de presse sur un sit...,Non structurées,"Les articles sont du texte libre, donc ils ne ..."
3,Données d'inventaire dans une base de données ...,Structurées,Les données sont stockées dans des tables avec...
4,Enregistrements d'entretiens issus d'une étude...,Non structurées,Les entretiens sont souvent des fichiers audio...


Exercise 7

In [ ]:
conversion_donnees = pd.DataFrame({
    "Source de données non structurées": [
        "Une série d'articles de blog sur les expériences de voyage",
        "Enregistrements audio des appels au service client",
        "Notes manuscrites issues d'une séance de brainstorming",
        "Un tutoriel vidéo sur la cuisine"
    ],
    "Méthode de conversion en données structurées": [
        "Utiliser le traitement automatique du langage naturel (NLP) pour extraire les lieux, dates, activités, avis et sentiments.",
        "Transcrire les audios en texte avec un outil de reconnaissance vocale, puis extraire les informations importantes.",
        "Numériser les notes avec un scanner ou une photo, utiliser l'OCR pour convertir l'écriture en texte, puis classer les idées par thème.",
        "Extraire la transcription de la vidéo, identifier les étapes, ingrédients, durées et actions, puis organiser ces informations dans un tableau."
    ],
    "Données structurées possibles": [
        "Destination, date du voyage, activité, avis positif/négatif, budget, moyen de transport",
        "Nom du client, date de l'appel, problème signalé, solution proposée, niveau de satisfaction",
        "Idée, catégorie, priorité, responsable, date, statut",
        "Nom de la recette, ingrédient, quantité, temps de cuisson, étape, matériel utilisé"
    ],
    "Justification": [
        "Les articles de blog sont du texte libre. Le NLP permet de transformer ce texte en informations organisées et exploitables.",
        "Les appels audio ne sont pas directement analysables sous forme de tableau. La transcription permet de transformer la voix en texte structuré.",
        "Les notes manuscrites sont difficiles à analyser automatiquement. L'OCR permet de convertir l'image en texte, puis de structurer les idées.",
        "Une vidéo contient de l'image et du son. La transcription et l'extraction des éléments clés permettent de créer un dataset utilisable."
    ]
})

conversion_donnees

,Source de données non structurées,Méthode de conversion en données structurées,Données structurées possibles,Justification
0,Une série d'articles de blog sur les expérienc...,Utiliser le traitement automatique du langage ...,"Destination, date du voyage, activité, avis po...",Les articles de blog sont du texte libre. Le N...
1,Enregistrements audio des appels au service cl...,Transcrire les audios en texte avec un outil d...,"Nom du client, date de l'appel, problème signa...",Les appels audio ne sont pas directement analy...
2,Notes manuscrites issues d'une séance de brain...,Numériser les notes avec un scanner ou une pho...,"Idée, catégorie, priorité, responsable, date, ...",Les notes manuscrites sont difficiles à analys...
3,Un tutoriel vidéo sur la cuisine,"Extraire la transcription de la vidéo, identif...","Nom de la recette, ingrédient, quantité, temps...",Une vidéo contient de l'image et du son. La tr...


Exercise 8

In [ ]:

url = "COLLE_ICI_LE_LIEN_RAW_DU_FICHIER_train.csv"

df = pd.read_csv(url)

df.head()

Exercise 9

In [ ]:

data = {
    "Nom": ["Ali", "Mariam", "Koffi", "Awa"],
    "Age": [22, 25, 20, 24],
    "Ville": ["Abidjan", "Bouaké", "Yamoussoukro", "Daloa"],
    "Note": [15, 17, 14, 16]
}

df = pd.DataFrame(data)

print("DataFrame créé :")
print(df)

df.to_excel("students.xlsx", index=False)

df.to_json("students.json", orient="records", indent=4, force_ascii=False)

print("Export terminé : fichiers students.xlsx et students.json créés.")

DataFrame créé :
      Nom  Age         Ville  Note
0     Ali   22       Abidjan    15
1  Mariam   25        Bouaké    17
2   Koffi   20  Yamoussoukro    14
3     Awa   24         Daloa    16
Export terminé : fichiers students.xlsx et students.json créés.


Exercise 10

In [ ]:

url = "C:\Users\ouedr\Desktop\TTA-ISMAEL\DI-bootcamp\Week2\Day1\ExerciseXP\Data\posts.json"

df = pd.read_json(url)
df.head()

,Nom,Age,Ville,Note
0,Ali,22,Abidjan,15
1,Mariam,25,Bouaké,17
2,Koffi,20,Yamoussoukro,14
3,Awa,24,Daloa,16
